client/src/commands/associateIdentityWithNode.ts

```diff
@@ -0,0 +1,188 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+'use strict';
+import { VSCodeBlockchainOutputAdapter } from '../logging/VSCodeBlockchainOutputAdapter';
+import { LogType } from '../logging/OutputAdapter';
+import { IBlockchainQuickPickItem, UserInputUtil } from './UserInputUtil';
+import { FabricWalletRegistryEntry } from '../fabric/FabricWalletRegistryEntry';
+import { FabricRuntimeUtil } from '../fabric/FabricRuntimeUtil';
+import { FabricEnvironmentRegistry } from '../fabric/FabricEnvironmentRegistry';
+import { FabricEnvironmentRegistryEntry } from '../fabric/FabricEnvironmentRegistryEntry';
+import { FabricEnvironment } from '../fabric/FabricEnvironment';
+import { FabricNode, FabricNodeType } from '../fabric/FabricNode';
+import { IFabricWallet } from '../fabric/IFabricWallet';
+import * as vscode from 'vscode';
+import { ExtensionCommands } from '../../ExtensionCommands';
+import { FabricEnvironmentTreeItem } from '../explorer/runtimeOps/disconnectedTree/FabricEnvironmentTreeItem';
+import { FabricCertificateAuthorityFactory } from '../fabric/FabricCertificateAuthorityFactory';
+import { IFabricCertificateAuthority } from '../fabric/IFabricCertificateAuthority';
+import { IFabricWalletGenerator } from '../fabric/IFabricWalletGenerator';
+import { FabricWalletGeneratorFactory } from '../fabric/FabricWalletGeneratorFactory';
+
+export async function associateIdentityWithNode(environmentRegistryEntry: FabricEnvironmentRegistryEntry, node: FabricNode): Promise<any> {
+    const outputAdapter: VSCodeBlockchainOutputAdapter = VSCodeBlockchainOutputAdapter.instance();
+    outputAdapter.log(LogType.INFO, undefined, 'associate identity with node');
+
+    try {
+        if (!environmentRegistryEntry || !node) {
+            // If called from command palette
+            const environemts: Array<FabricEnvironmentRegistryEntry> = FabricEnvironmentRegistry.instance().getAll();
+            if (environemts.length === 0) {
+                outputAdapter.log(LogType.ERROR, `Add an environement to associate nodes with identities. ${FabricRuntimeUtil.LOCAL_FABRIC} cannot be editted.`);
```

Comment:
Can you make this:
`Add an environment to associate identities with nodes`

Also, will need to replace `environement` with `environment` - I think there might be a few cases of this over the PR and potentially the remote-deploy branch